packages/adapters/eigenda/src/error.rs

```diff
@@ -0,0 +1,25 @@
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("transport error: {0}")]
+    Transport(#[from] tonic::transport::Error),
+    #[error("RPC error: {0}")]
+    Rpc(#[from] tonic::Status),
+    #[error("authentication failed")]
+    AuthenticationFailed,
+    #[error("EigenDA client error: {0}")]
+    EigenDAClient(#[from] rust_eigenda_client::errors::EigenClientError),
+    #[error("other error: {0}")]
+    Other(String),
```

Comment:
Not a fan of these catch all error variants. Would be good to remove this in favour of using specific error types.

Reply:
addressed in [e2ce980](https://github.com/FuelLabs/fuel-block-committer/pull/187/commits/e2ce9806226d109940ad6de7218ba01ed440584b), but is an ongoing effort